pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -572,3 +598,51 @@ func getChaosType() error {
 	}
 	return nil
 }
+
+// reconcileForDelete
+func (r *ReconcileChaosEngine) reconcileForDelete(request reconcile.Request) (reconcile.Result, error) {
+
+	r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "Stopping reconcile for chaosEngine", "Will remove all Chaos resources for chaosEngineUUID: %v", string(engine.Instance.UID))
+	optsDelete := []client.DeleteAllOfOption{
+		client.InNamespace(request.NamespacedName.Namespace),
+		client.MatchingLabels{"engineUUID": string(engine.Instance.UID)},
+	}
+	reqLogger := chaosTypes.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	err := r.client.DeleteAllOf(context.TODO(), &corev1.Pod{}, optsDelete...)
+	if err != nil {
+		reqLogger.Error(err, "Unable to delete chaosEngine allocated Chaos Resources")
```

Comment:
return the error

Reply:
Returning just below.

Reply:
This log should not be in the child function, You can form the error and return it the parent function will print it

Reply:
Parent function only will inform that the function block was unable to do its tasks, but inner logging will take us to the line where the error occured, seems good to me @chandankumar4

Reply:
Btw in this case, I'm passing the responsibility to `reconcileForDelete` to return the reconciler, so error handling has to be done inside the function only.

Reply:
According to the above statement, you are printing the error here itself then why the error is also returning.
I think it's a repetitive error printing!!